frontend/src/concepts/modelRegistry/apiHooks/__tests__/useModelRegistryServices.spec.ts

```diff
@@ -22,102 +25,164 @@ jest.mock('~/concepts/modelRegistry/apiHooks/useModelRegistryServices', () => ({
   fetchServices: jest.fn(),
 }));
 
+jest.mock('~/api/useRulesReview', () => ({
+  useRulesReview: jest.fn(),
+}));
+
+jest.mock('~/api/useAccessReview');
+jest.mock('~/api/useRulesReview');
+const objectToStandardUseFetchState = (obj: ModelRegistryServicesResult) => [
+  obj.modelRegistryServices,
+  obj.isLoaded,
+  obj.error,
+  obj.refreshRulesReview,
+];
+
 const mockGetResource = jest.mocked(k8sGetResource<ServiceKind>);
 const mockListServices = jest.mocked(listServices);
 
 describe('useModelRegistryServices', () => {
   beforeEach(() => {
     jest.clearAllMocks();
+    (useAccessReview as jest.Mock).mockReturnValue([false, true]);
+    (useRulesReview as jest.Mock).mockReturnValue([{ resourceRules: [] }, true, jest.fn()]);
   });
+  it('should return loading state initially', () => {
+    const { result } = testHook(() => useModelRegistryServices('namespace'))();
 
-  test('returns loading state initially', () => {
-    (useAccessReview as jest.Mock).mockReturnValue([false, false]);
-    (useRulesReview as jest.Mock).mockReturnValue([{}, false]);
-    const renderResult = testHook(useModelRegistryServices)();
+    const [modelRegistryServices, isLoaded, error] = objectToStandardUseFetchState(result.current);
 
-    expect(renderResult.result.current[1]).toBe(false);
-    expect(renderResult.result.current[0]).toEqual([]);
+    expect(modelRegistryServices).toEqual([]);
+    expect(isLoaded).toBe(false);
+    expect(error).toBeUndefined();
+  });
+  it('returns services fetched by names when allowList is false', async () => {
+    (useAccessReview as jest.Mock).mockReturnValue([false, true]);
+    (useRulesReview as jest.Mock).mockReturnValue([
+      {
+        resourceRules: [
+          {
+            resources: ['services'],
+            verbs: ['get'],
+            resourceNames: ['service-1', 'service-2'],
+          },
+        ],
+      },
+      true,
+      jest.fn(),
+    ]);
+    mockGetResource.mockResolvedValueOnce(
+      mockModelRegistryService({ name: 'service-1', namespace: 'test-namespace' }),
+    );
+    mockGetResource.mockResolvedValueOnce(
+      mockModelRegistryService({ name: 'service-2', namespace: 'test-namespace' }),
+    );
+
+    const { result } = testHook(() => useModelRegistryServices('test-namespace'))();
+
+    await act(async () => {
```

Comment:
i wonder if the uses of `act` like this could be replaced with something like:

    await waitFor(() => {
        const [, isLoaded] = objectToStandardUseFetchState(result.current);
        expect(isLoaded).toBe(true);
    });
    const [services, isLoaded] = objectToStandardUseFetchState(result.current);

Reply:
Those tests using waitFor, were there before and I really had some issues with them, so I just took this approach

Reply:
But I can try your suggestion and see if it works

Reply:
there are several such uses of `act`, it just popped out to me because of the deprecated strikethrough in my IDE 😆 

but more of a nit i suppose so if it's a lot of effort to address i'm okay approving.